test/jasmine/tests/gl3d_plot_interact_test.js

```diff
@@ -903,16 +1164,117 @@ describe('Test gl3d drag and wheel interactions', function() {
             return scroll(sceneTarget);
         })
         .then(function() {
+            _assertAndReset(1);
+
             return scroll(sceneTarget2);
         })
         .then(function() {
-            _assertAndReset(2);
+            _assertAndReset(1);
         })
         .catch(failTest)
         .then(done);
     });
 
-    it('@gl should fire plotly_relayouting events', function(done) {
+    it('@gl should fire plotly_relayouting events when dragged - perspective case', function(done) {
+        var sceneTarget, relayoutEvent;
+
+        var nsteps = 10;
+        var relayoutCnt = 0;
+        var events = [];
+
+        var mock = {
+            data: [
+                { type: 'scatter3d', x: [1, 2, 3], y: [2, 3, 1], z: [3, 1, 2] }
+            ],
+            layout: {
+                scene: { camera: { projection: {type: 'perspective'}, eye: { x: 0.1, y: 0.1, z: 1 }}},
+                width: 400, height: 400
+            }
+        };
+
+        Plotly.plot(gd, mock)
+        .then(function() {
+            gd.on('plotly_relayout', function(e) {
+                relayoutCnt++;
+                relayoutEvent = e;
+            });
+            gd.on('plotly_relayouting', function(e) {
+                events.push(e);
+            });
+
+            sceneTarget = gd.querySelector('.svg-container .gl-container #scene canvas');
+
+            return drag({
+                node: sceneTarget,
+                pos0: [200, 200],
+                posN: [100, 100],
+                nsteps: nsteps,
+                buttons: 1,
+                noCover: true
+            });
+        })
+        .then(function() {
+            expect(events.length).toEqual(nsteps);
+            expect(relayoutCnt).toEqual(1);
+            Object.keys(relayoutEvent).sort().forEach(function(key) {
+                expect(Object.keys(events[0])).toContain(key);
+            });
```

Comment:
Can you also check that this set of relayout event keys does not include `scene.aspectratio`?

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4292/commits/3b28faef2ddd72d1ce6f6d2812f5b1f01917075f. I also added few more tests.